core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -4728,7 +4728,7 @@ public ValueMetaInterface getValueFromSQLType( DatabaseMeta databaseMeta, String
 
           if ( databaseMeta.getDatabaseInterface() instanceof OracleDatabaseMeta ) {
             if ( precision == 0 && length == 38 ) {
-              valtype = ValueMetaInterface.TYPE_INTEGER;
+              valtype = databaseMeta.strictBigNumberInterpretation() ? TYPE_BIGNUMBER : TYPE_INTEGER;
```

Comment:
You already know it's a OracleDatabaseMeta otherwise you would've not entered this conditional branch. Cast databaseMeta to OracleDatabaseMeta so you can call the method.